<launch>
  <arg name="esdf_path"        default="$(find kimera_scene_graph)/vxblx_files/all_tesse_skeleton_4.vxblx" />
  <!--
  <arg name="scene_graph_path" default="$(find kimera_scene_graph)/vxblx_files/all_tesse_sparse_graph_4.vxblx" />
  -->
  <arg name="scene_graph_path" default=""/>
  <arg name="sparse_graph_path" default="$(find kimera_scene_graph)/vxblx_files/all_tesse_sparse_graph_4.vxblx" />

  <arg name="frame_id" default="odom" />

  <group ns="firefly">
    <node name="kimera_path_planning_node" pkg="kimera_path_planning" type="kimera_path_planning_node" output="screen"
      launch-prefix="gdb -ex run --args"
      args="-alsologtostderr -colorlogtostderr -logtostderr"
      clear_params="true">
      <!-- Data paths -->
      <param name="esdf_path"         value="$(arg esdf_path)" />
      <param name="scene_graph_path"  value="$(arg scene_graph_path)" />
      <param name="sparse_graph_path" value="$(arg sparse_graph_path)" />


      <!-- Start params for Scene-Graph Reconstruction -->
      <!-- Should not be needed if we were to load the scene-graph -->

      <!-- Object Finder Type: 0: region growing, 1: euclidean clustering -->
      <param name="object_finder_type"                 value="1"/>
      <param name="room_finder_esdf_slice_level"       value="3.2"/>

      <!-- The only dynamic label we have right now are humans, with label 20 -->
      <rosparam param="dynamic_semantic_labels">[]</rosparam>
      <!-- Stuff are: walls (19), ceiling and floor (3) -->
      <rosparam param="stuff_labels">[19, 3]</rosparam>
      <rosparam param="walls_labels">[19, 12]</rosparam>
      <rosparam param="floor_labels">[3, 1, 4]</rosparam>

      <param name="load_tsdf_map"     value="$(find kimera_scene_graph)/vxblx_files/all_tesse_tsdf.vxblx"/>
      <param name="load_esdf_map"     value="$(find kimera_scene_graph)/vxblx_files/all_tesse_esdf.vxblx"/>
      <param name="load_sparse_graph" value="$(find kimera_scene_graph)/vxblx_files/all_tesse_sparse_graph_2.vxblx"/>

      <!-- Whether to rebuild the esdf layer from scratch -->
      <param name="build_esdf_batch" value="false"/>

      <param name="semantic_label_2_color_csv_filepath"
      value="$(find kimera_semantics_ros)/cfg/tesse_multiscene_office1_segmentation_mapping.csv"/>
      <!-- End params for Scene-Graph Reconstruction -->

      <param name="visualize" value="false" />
      <param name="robot_radius" value="0.05" />
      <param name="sampling_dt"  value="0.1" />
      <param name="tsdf_voxel_size" value="0.05" />
      <param name="tsdf_voxels_per_side" value="32" />

      <param name="esdf_map_out"  value="esdf_map_out" />
      <param name="tsdf_map_out"  value="tsdf_map_out" />

      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="frame_id" value="$(arg frame_id)" />
      <param name="world_frame" value="$(arg frame_id)" />
      <param name="trust_approx_solution" value="true" />
      <param name="publish_traversable" value="true" />
      <param name="split_at_collisions" value="true" />
      <param name="optimize_time" value="true" />
      <param name="verbose" value="true" />
    </node>
  </group>
</launch>
